usr/local/sbin/resticw

```diff
@@ -0,0 +1,141 @@
+error "${longs[$match]} must not have an argument"; fi
+elif [[ $value = false ]]; then
+if [[ ${#argv[@]} -eq 0 || ${argv[0]} = '--' ]]; then
+error "${long} requires argument"; fi; value=${argv[0]}; argv=("${argv[@]:1}")
+fi; if [[ $value = false ]]; then value=true; fi; fi; parsed_params+=("$match")
+parsed_values+=("$value"); }; required() { local initial_left=("${left[@]}")
+local node_idx; ((testdepth++)) || true; for node_idx in "$@"; do
+if ! "node_$node_idx"; then left=("${initial_left[@]}"); ((testdepth--)) || true
+return 1; fi; done; if [[ $((--testdepth)) -eq 0 ]]; then
+left=("${initial_left[@]}"); for node_idx in "$@"; do "node_$node_idx"; done; fi
+return 0; }; optional() { local node_idx; for node_idx in "$@"; do
+"node_$node_idx"; done; return 0; }; value() { local i
+for i in "${!left[@]}"; do local l=${left[$i]}
+if [[ ${parsed_params[$l]} = "$2" ]]; then
+left=("${left[@]:0:$i}" "${left[@]:((i+1))}")
+[[ $testdepth -gt 0 ]] && return 0; local value
+value=$(printf -- "%q" "${parsed_values[$l]}"); if [[ $3 = true ]]; then
+eval "var_$1+=($value)"; else eval "var_$1=$value"; fi; return 0; fi; done
+return 1; }; stdout() { printf -- "cat <<'EOM'\n%s\nEOM\n" "$1"; }; stderr() {
+printf -- "cat <<'EOM' >&2\n%s\nEOM\n" "$1"; }; error() {
+[[ -n $1 ]] && stderr "$1"; stderr "$usage"; _return 1; }; _return() {
+printf -- "exit %d\n" "$1"; exit "$1"; }; set -e; trimmed_doc=${DOC:0:596}
+usage=${DOC:222:47}; digest=5f4ba; shorts=(-p); longs=(--profile); argcounts=(1)
+node_0(){ value __profile 0; }; node_1(){ value _restic_arguments_ a; }
+node_2(){ optional 0; }; node_3(){ optional 2; }; node_4(){ required 3 1; }
+node_5(){ required 4; }; cat <<<' docopt_exit() {
+[[ -n $1 ]] && printf "%s\n" "$1" >&2; printf "%s\n" "${DOC:222:47}" >&2; exit 1
+}'; unset var___profile var__restic_arguments_; parse 5 "$@"
+local prefix=${DOCOPT_PREFIX:-''}; unset "${prefix}__profile" \
+"${prefix}_restic_arguments_"
+eval "${prefix}"'__profile=${var___profile:-default}'
+eval "${prefix}"'_restic_arguments_=${var__restic_arguments_:-}'
+local docopt_i=1; [[ $BASH_VERSION =~ ^4.3 ]] && docopt_i=2
+for ((;docopt_i>0;docopt_i--)); do declare -p "${prefix}__profile" \
+"${prefix}_restic_arguments_"; done; }
+# docopt parser above, complete command for generating this parser is `docopt.sh resticw`
+
+# Parse arguments
+eval "$(set +u; docopt "$@")" # See https://github.com/andsens/docopt.sh for the magic :)
+
+# --^^^-- END OF GENERATED COMMAND LINE PARSING STUFF --^^^--
+#
+# --vvv--             ACTUAL SCRIPT BELOW             --vvv--
+
+ENV_DIR=/etc/restic
+
+profile_file="${ENV_DIR}/${__profile}.env"
+
+[[ ! -f "$profile_file" ]] && echo "Invalid profile: No such environment file ${profile_file}" && exit 2
+
+if [ $(id -u) -ne 0 ]; then
+  echo "Error: Run it as root or use sudo, as the script needs to load the environment from protected file ${profile_file}"
+  exit 3
+fi
+
+echo -e "‣ Using profile: ${__profile}  --  (${profile_file})\n"
```

Comment:
Tip: use `printf(1)` for more advanced formatting.

```bash
printf "‣ Using profile: %s -- (%s)\n" "${__profile}" "${profile_file}"
```

Reply:
Hey, thanks! I can't spot much of a difference, the `printf` version being even longer. Do you want to change this?

I find the string interpolation in echo much easier to read than using `%format` and passing actual arguments at the end. It is not using any advanced formating actually, hehe.

Reply:
Preferences of style :). I see benefits in both. No need to fix, just a tip